src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -328,23 +328,42 @@ an request on an object (not a container) that has a trailing slash will lead
       }
 
       if (contentLength == 0) { // Check if directory
-        if (contentType.equals("application/directory")) {
+        if (contentType.equals("application/directory")
+                || contentType.equals("application/octet-stream")) {
           isDirectory = true;
-        } else if (!contentType.equals("application/octet-stream")) {
+        /*} else if (contentType.equals("application/octet-stream")) {
+          FileStatus[] listing = list(hostName, path, false, true);
+          if (listing.length > 1) {
+            // If additional objects are returned then the object marks a directory
+            isDirectory = true;
+          }
+        }*/
+        } else {
```

Comment:
@gilv I was unsure if this logic would be used or not. I ended up using it but refactored the directory check into a separate method.